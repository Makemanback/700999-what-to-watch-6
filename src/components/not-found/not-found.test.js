import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import {screen, render} from '@testing-library/react';
import NotFound from './not-found';

test(`Should NotFound render correctly`, () => {
  const history = createMemoryHistory();

  render(
      <Router history={history}>
        <NotFound />
      </Router>
  );

  expect(screen.getByText(/Not found/i)).toBeInTheDocument();
  expect(screen.getByText(/Go to the main page/i)).toBeInTheDocument();
});
