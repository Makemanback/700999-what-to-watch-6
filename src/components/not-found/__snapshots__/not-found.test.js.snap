// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should NotFound render correctly 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        class="page-title user-page__title"
      >
        Not found
      </h1>
    </header>
    <div
      class="not-found"
    >
      <a
        class="not-found__link"
        href="/"
      >
        Go to the main page
      </a>
    </div>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
